client/src/views/Dataset.vue

```diff
@@ -184,55 +166,38 @@ export default {
       return Promise.resolve(true);
     },
     async save() {
-      var user = this.girderRest.user;
-      var initial =
-        user.firstName.charAt(0).toLocaleUpperCase() +
-        user.lastName.charAt(0).toLocaleUpperCase();
-      var date = new Date().toISOString().slice(0, 10);
-      var note = "";
       if (this.newNote.trim()) {
-        note =
-          (this.note ? this.note + "\n" : "") +
-          `${initial}(${date}): ${this.newNote}`;
-      } else {
-        note = this.note;
+        await this.djangoRest.addScanNote(this.currentSession.id, this.newNote);
+        this.newNote = "";
       }
-      var meta = {
-        ...this.currentSession.meta,
-        ...{
-          note,
-          rating: this.rating !== undefined ? this.rating : null,
-          reviewer: user.firstName + " " + user.lastName
-        }
-      };
-      await this.girderRest.put(
-        `folder/${this.currentSession.folderId}/metadata?allowNull=true`,
-        meta
-      );
-      this.newNote = "";
-      this.currentSession.meta = meta;
-      this.reviewer = meta.reviewer;
-      this.reviewChanged = false;
-      this.noteLastSaveTime = new Date().toISOString();
+      if (this.decisionChanged) {
+        await this.djangoRest.setDecision(
+          this.currentSession.id,
+          this.decision
+        );
+        this.decisionChanged = false;
+      }
+      this.reloadScan();
     },
     enableEditHistroy() {
       this.editingNoteDialog = true;
-      this.editingNote = this.note;
+      this.editingNotes = [];
+      for (let i = 0; i < this.notes.length; i++) {
+        let note = this.notes[i];
+        this.editingNotes.push(note.note);
+      }
     },
     async saveNoteHistory() {
-      var meta = {
-        ...this.currentSession.meta,
-        ...{
-          note: this.editingNote
+      for (let i = 0; i < this.editingNotes.length; i++) {
+        let editingNote = this.editingNotes[i];
+        let note = this.notes[i];
+        if (editingNote !== note.note) {
+          await this.djangoRest.setScanNote(note.id, editingNote);
```

Comment:
We can avoid the unneeded await-in-loop here.

```javascript
const modified = this.notes.filter((note, i) => note.note !== this.editingNotes[i]);
```

Then we can await a `Promise.all` of XHRs for the resulting array.